use-cases/NVIDIA-GPU-passthrough-and-Kata.md

```diff
@@ -0,0 +1,296 @@
+/usr/share/kata-containers/vmlinux-gpu.container -> vmlinux-4.19.86-69-gpu
+/usr/share/kata-containers/vmlinuz-gpu.container -> vmlinuz-4.19.86-69-gpu
+```
+
+To build NVIDIA Driver in kata container, `kernel-devel` is required.  
+Here is a way to get the rpm package for `kernel-devel`
+```
+$ cd kata-linux-4.19.86-68
+$ make rpm-pkg
+Output RPMs:
+~/rpmbuild/RPMS/x86_64/kernel-devel-4.19.86_gpu-1.x86_64.rpm
+```
+
+> **kernel-devel-4.19.86_gpu-1.x86_64.rpm** should be installed in kata container before run Nvidia driver installer.
+
+
+Please update the kernel parameter in `configuration.toml` before use the new guest kernel.
+```
+kernel = "/usr/share/kata-containers/vmlinuz-gpu.container"
+```
+
+## NVIDIA GPU Pass-Through mode with Kata Containers
+Use the following steps to pass a Nvidia GPU device in Pass-Through mode with Kata:
+
+1. Find the Bus-Device-Function (BDF) for GPU device on host:
+```
+$ sudo lspci -nn -D | grep -i nvidia
+0000:04:00.0 3D controller [0302]: NVIDIA Corporation Device [10de:15f8] (rev a1)
+0000:84:00.0 3D controller [0302]: NVIDIA Corporation Device [10de:15f8] (rev a1)
+```
+
+2. Run the previous command to determine the BDF for the GPU device on host.
+From the previous output, PCI address `0000:04:00.0` is assigned to the hardware GPU device.
+This BDF is used later to unbind the GPU device from the host.
+`10de:15f8` is the device ID of the hardware GPU device. It is used later to rebind the GPU device to `vfio-pci` driver.
+2. Find the IOMMU group for the GPU device:
+```
+$ BDF="0000:04:00.0"
+$ readlink -e /sys/bus/pci/devices/$BDF/iommu_group
+/sys/kernel/iommu_groups/45
+```
+The previous output shows that the GPU belongs to IOMMU group 45.
+
+4. Check the IOMMU group number under `/dev/vfio`:
+```
+$ ls -l /dev/vfio
+ total 0
+crw------- 1 root root 248,   0 Feb 28 09:57 45
+crw------- 1 root root 248,   1 Feb 28 09:57 54
+crw-rw-rw- 1 root root  10, 196 Feb 28 09:57 vfio
+```
+
+5. Start a Kata container with GPU device:
+```
+sudo docker run -it --runtime=kata-runtime --rm --device /dev/vfio/45 centos /bin/bash
+```
+
+6. Run `lspci` within the container to verify the GPU device is seen in the list of the PCI devices. Note the vendor-device id of the GPU (`10de:15f8`) in the `lspci` output.
+```
+$ lspci -nn -D | grep '10de:15f8'
+0000:01:01.0 3D controller [0302]: NVIDIA Corporation GP100GL [Tesla P100 PCIe 16GB] [10de:15f8] (rev a1)
+```
+
+7. Additionally, you can check the BAR region for the Nvidia GPU device:
+```
+$ lspci -s 01:01.0 -vv | grep Region
+	Region 0: Memory at c0000000 (32-bit, non-prefetchable) [disabled] [size=16M]
+	Region 1: Memory at 4400000000 (64-bit, prefetchable) [disabled] [size=16G]
+	Region 3: Memory at 4800000000 (64-bit, prefetchable) [disabled] [size=32M]
+```
+> If you see a message similar to the above, the BAR space of the Nvidia GPU has been successfully allocated.
+
+## NVIDIA vGPU mode with Kata Containers
+> TODO
```

Comment:
Heh :-) We'll have to put some more verbose sentence in here before we can merge this PR :-)

Reply:
Added a note.